client/keystore/src/local.rs

```diff
@@ -420,15 +420,15 @@ impl KeystoreInner {
 	/// Write the given `data` to `file`.
 	fn write_to_file(file: PathBuf, data: &str) -> Result<()> {
 		let mut file = File::create(file)?;
-		serde_json::to_writer(&file, data)?;
-		file.flush()?;
 
 		#[cfg(target_family = "unix")]
 		{
 			use std::os::unix::fs::PermissionsExt;
 			file.set_permissions(fs::Permissions::from_mode(0o600))?;
 		}
 
+		serde_json::to_writer(&file, data)?;
+		file.flush()?;
```

Comment:
Why this change here?

Reply:
Ooops. Forgot it in this changeset. Was thinking about if this reduces the race-condition that another process opens the file before the permissions are applied. It is a very thin chance, and did not want to push it :worried: